### PR TITLE
[Storage] Upgrade API version to 2022-09-01

### PR DESCRIPTION
**Pull request by yifanz0**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/storage/Azure.ResourceManager.Storage/api/Azure.ResourceManager.Storage.netstandard2.0.cs [hunk 1]
```diff
@@ -1126,6 +1130,23 @@ public partial class EncryptionScopeKeyVaultProperties
         public override string ToString() { throw null; }
     }
     [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct FailoverType : System.IEquatable<Azure.ResourceManager.Storage.Models.FailoverType>
```
Comment (ArcturusZhang) [inline sdk/storage/Azure.ResourceManager.Storage/api/Azure.ResourceManager.Storage.netstandard2.0.cs:1133]: Can we rename this to `StorageAccountFailoverType`?

File: sdk/storage/Azure.ResourceManager.Storage/api/Azure.ResourceManager.Storage.netstandard2.0.cs [hunk 2]
```diff
@@ -1412,6 +1433,25 @@ public partial class LegalHoldTag
         public System.DateTimeOffset? Timestamp { get { throw null; } }
         public string Upn { get { throw null; } }
     }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct ListEncryptionScopesInclude : System.IEquatable<Azure.ResourceManager.Storage.Models.ListEncryptionScopesInclude>
```
Comment (ArcturusZhang) [inline sdk/storage/Azure.ResourceManager.Storage/api/Azure.ResourceManager.Storage.netstandard2.0.cs:1437]: Could we rename this to `EncryptionScopesIncludeType`?
Only suggestion, please feel free to change it to names you feel reasonable

File: sdk/storage/Azure.ResourceManager.Storage/api/Azure.ResourceManager.Storage.netstandard2.0.cs (hunk 1 above)
Comment (yifanz0) [reply sdk/storage/Azure.ResourceManager.Storage/api/Azure.ResourceManager.Storage.netstandard2.0.cs:1133]: Done

File: sdk/storage/Azure.ResourceManager.Storage/api/Azure.ResourceManager.Storage.netstandard2.0.cs (hunk 2 above)
Comment (yifanz0) [reply sdk/storage/Azure.ResourceManager.Storage/api/Azure.ResourceManager.Storage.netstandard2.0.cs:1437]: Done

**Review by Yao725 — APPROVED** [review]

**Pull request merged**